### PR TITLE
Adds support for aws_wafv2_web_acl_association in CKV2_AWS_29

### LINKED ISSUE #2367: CKV2_AWS_29 should support aws_wafv2_web_acl_association
The check "Ensure public API gateway are protected by WAF" only supports the association if it is done using the `aws_wafregional_web_acl_association`. Considering that the rule description in bridgecrew mentions WAFv2, it should support `aws_wafv2_web_acl_association`.

https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/graph_checks/aws/APIProtectedByWAF.yaml#L36

### PR DESCRIPTION
**Pull request by BenjaDiaz**
Fixes #2367

### CODE REVIEW

File: tests/terraform/graph/checks/resources/APIProtectedByWAF/main.tf [hunk 1]
```diff
@@ -62,7 +62,18 @@ resource "aws_api_gateway_stage" "pass" {
   stage_name    = "example"
 }
 
+resource "aws_api_gateway_stage_wafv2" "pass" {
```
Comment (gruebel) [inline tests/terraform/graph/checks/resources/APIProtectedByWAF/main.tf:65]:
```suggestion
resource "aws_api_gateway_stage" "wafv2_pass" {
```
that's what you probably meant, right?

**Review by gruebel — COMMENTED** [review]
and then you need to adjust the name for the references

File: tests/terraform/graph/checks/resources/APIProtectedByWAF/main.tf (hunk 1 above)
Comment (gruebel) [reply tests/terraform/graph/checks/resources/APIProtectedByWAF/main.tf:65]: and don't forget to change it in `expected.yml` file 😄

Comment (BenjaDiaz) [reply tests/terraform/graph/checks/resources/APIProtectedByWAF/main.tf:65]: Hahaha, you are totally right. Sorry! Fixing it now.

**Review by gruebel — APPROVED** [review]
Thanks for the contribution 🎉

**Pull request merged**